Dockerfile

```diff
@@ -31,9 +31,17 @@ RUN apt-get update && apt-get -y install\
 
 RUN pip3 install --user py-solc
 
-# install solc
-RUN wget https://github.com/ethereum/solidity/releases/download/v0.4.24/solc-static-linux -O /usr/local/bin/solc &&\
-  chmod u+x /usr/local/bin/solc
+# install pip and py-solc
+RUN apt-get update && apt-get install -y python3-pip
+
+RUN pip3 install --user py-solc termcolor
+
+# install truffle for project compilation
+RUN apt-get update && apt-get install -y nodejs\
+                npm
+
+# install truffle v5-beta
+RUN npm install -g truffle@beta
```

Comment:
Maybe better to default to "latest", while allowing to change it easily, with:
`ARG truffle="latest"`

Reply:
Does `"latest"` include beta versions? Because I was told to explicitly use v5 since apparently there are quite some changes at this moment.

Reply:
No it does not include beta versions, but Solidity projects are more likely to use a stable version, or at least they should. The problem with beta version is that they can break unexpectedly and if this causes Securify to fail in turn, we may end up wasting time finding out where this comes from.

Reply:
I agree, it's just that Hubert told me to use v5 explicitly. I'll will see if latest works too and change it. To be fair, the only consistency we require from truffle is the naming of the keys in the compilation output, which did not change in v5 I think. I will test and change this.